openhtf/plugs/uart.py

```diff
@@ -0,0 +1,331 @@
+# Copyright 2014 Google Inc. All Rights Reserved.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""SerialPlug used for serial communication.
+
+  Call SerialPlug's GetsSerialHandle to get an instance
+  to each serial device you need access to.
+
+  SerialPlug.GetSerialHandle parameters:
+    serial_port: /dev/ttyUSB0     # serial dev name.
+    serial_baudrate: 19600        # baud rate
+    serial_parity: N              # PARITY_NONE, PARITY_EVEN, PARITY_ODD,
+                                    PARITY_MARK,
+                                    PARITY_SPACE = 'N', 'E', 'O', 'M', 'S'
+    serial_stopbits: 1            # STOPBITS_ONE, STOPBITS_ONE_POINT_FIVE,
+                                    STOPBITS_TWO = (1, 1.5, 2)
+    serial_bytesize: 8            # FIVEBITS, SIXBITS, SEVENBITS,
+                                    EIGHTBITS = (5, 6, 7, 8)
+    serial_rtscts: 0              # boolean type
+    serial_timeout: 0
+    serial_min_buffer_time: 50    # minimum wait time after a write before
+                                    we read the input buffer
+"""
+import binascii
+import logging
+import time
+
+import openhtf.conf as conf
+import openhtf.plugs as plugs
+from openhtf.util import timeouts
+import serial
+
+conf.Declare('SerialPlug')
```

Comment:
I think we've decided to standardize config key names to use underscore_syntax instead of CamelCase.